src/plugins/mf_priority.cpp

```diff
@@ -1116,6 +1144,42 @@ extern "C" int flux_plugin_init (flux_plugin_t *p)
     priority_weights["queue"] = DEFAULT_QUEUE_WEIGHT;
     priority_weights["age"] = DEFAULT_AGE_WEIGHT;
 
+    // initialize the plugin with total node and core counts
+    flux_t *h;
+    flux_future_t *f;
+    char *core;
+
+    h = flux_jobtap_get_flux (p);
+    if (!(f = flux_kvs_lookup (h,
+                               NULL,
+                               FLUX_KVS_WATCH | FLUX_KVS_WAITCREATE,
+                               "resource.R"))) {
+        flux_log_error (h, "flux_kvs_lookup");
+        return -1;
+    }
+
+    if (flux_kvs_lookup_get_unpack (f,
+                                    "{s{s[{s{s:s}}]}}",
+                                    "execution",
+                                      "R_lite",
+                                        "children",
+                                          "core", &core) < 0) {
+        flux_log_error (h, "flux_kvs_lookup_unpack");
+        return -1;
+    }
```

Comment:
Normally I would request that this synchronous get be replaced with an asynchronous `flux_future_then(3)` and the parsing of R handled in a callback. However, it seems like the result is required for validation of jobs, and you probably don't want to let some jobs through erroneously while waiting for `resource.R`, so perhaps this is actually the right solution. The job manager will block here while waiting for R when `mf_priority.so` is loaded, but it _should_ be a very short time, and while this is occurring job management will also pause. Most of the time this will occur during job manager module load, which has other synchronous work anyway.

I'd at least suggest a comment here describing why a synchronous get is used in this case.

@garlick: any other thoughts?

Reply:
The FLUX_KVS_WATCH flag should be dropped.  That's only used when you want to receive a response for every change to the key, and here the future is being destroyed after the first response.

IRL, we only load `mf_priority.so` in the system instance so `resource.R` should exist already and the KVS lookup should be fast.   FLUX_KVS_WAITCREATE is probably is needed in test where the plugin is loaded in a test instance and  `resource.R` is dynamically discovered though.

Yes a comment would be good since synchronous activities always raise eyebrows.